be/src/vec/common/allocator.h

```diff
@@ -138,7 +145,7 @@ class Allocator {
     }
 
     /// Free memory range.
-    void free(void* buf, size_t size = -1) {
```

Comment:
Better not change it to static. May have some potential bugs.
For example, the free logic is not a common free logic, it will deal with mmap, chunk allocator. If the memory is not allocated from Allocator's alloc method then the free logic is wrong.
And we do not want to this usage: Allocator::free(). All data structures that want to use this allocator has to inherit Allocator.

Reply:
@yiguolei How can I use it like 'static void deleter(void* data) { Allocator<false, false, false>::free(data); }'.In the scenario of RPC, in order to achieve zero copy, I need to hand over the data to the append method for management, and need to define a collector that is static. If I don't do this, should I inherit the allocator?

Reply:
It will core, if the memory is not allocated by using Allocator's alloc method because it will deal mmap, chunk allocator inside the allocator. For example, if the memory is alloc by using malloc, its size maybe 128MB, and then you call Allocator::free, allocator will try use munmap to free the memory.
Could you please provide more detail info about the RPC's zero copy?

Reply:
The data used in rpc is composed of the ownsilce class, constructed by faststring. I will now change it to `static void deleter(void* data) { ::free(data); }`, it would work?